code/modules/events/disease_outbreak.dm

```diff
@@ -83,81 +127,249 @@
+		log_game("An event has triggered a random advanced virus outbreak on [key_name(victim)]! It has these symptoms: [english_list(name_symptoms)]. It is transmissable via [advanced_disease.spread_text].")
 		announce_to_ghosts(victim)
 	else
 		log_game("An event attempted to trigger a random advanced virus outbreak on [key_name(victim)], but failed.")
 
-/datum/event_admin_setup/disease_outbreak
-	///Admin selected disease, to be passed down to the round_event
-	var/virus_type
+/datum/disease/advance/random/event
+	name = "Experimental Disease"
+	copy_type = /datum/disease/advance
 
-/// Checks for candidates. Returns false if there isn't enough
-/datum/event_admin_setup/disease_outbreak/proc/candidate_check()
-	var/datum/round_event_control/disease_outbreak/disease_control = event_control
-	disease_control.generate_candidates() //can_spawn_event() is bypassed by admin_setup, so this makes sure that the candidates are still generated
-	return length(disease_control.disease_candidates)
+/datum/round_event/disease_outbreak/advance/setup()
+	announce_when = ADV_ANNOUNCE_DELAY
 
-/datum/event_admin_setup/disease_outbreak/prompt_admins()
-	var/candidate_count = candidate_check()
-	if(!candidate_check())
-		tgui_alert(usr, "There are no candidates eligible to recieve a disease!", "Error")
-		return ADMIN_CANCEL_EVENT
-	tgui_alert(usr, "[candidate_count] candidates found!", "Disease Outbreak")
+/**
+ * Generate advanced virus
+ *
+ * Uses the parameters to create a list of symptoms, picking from various severities
+ * Viral Evolution and Eternal Youth are special modifiers, so we roll separately.
+ */
 
-	if(tgui_alert(usr, "Select a specific disease?", "Sickening behavior", list("Yes", "No")) == "Yes")
-		virus_type = tgui_input_list(usr, "Warning: Some of these are EXTREMELY dangerous.","Bacteria Hysteria", subtypesof(/datum/disease))
+/datum/disease/advance/random/event/New(max_symptoms, max_severity)
+	var/list/datum/symptom/possible_symptoms = list(
+		/datum/symptom/beard,
+		/datum/symptom/chills,
+		/datum/symptom/choking,
+		/datum/symptom/confusion,
+		/datum/symptom/cough,
+		/datum/symptom/deafness,
+		/datum/symptom/dizzy,
+		/datum/symptom/fever,
+		/datum/symptom/headache,
+		/datum/symptom/itching,
+		/datum/symptom/shedding,
+		/datum/symptom/sneeze,
+		/datum/symptom/vomit,
+		/datum/symptom/weight_loss,
+	)
 
-/datum/event_admin_setup/disease_outbreak/apply_to_event(datum/round_event/disease_outbreak/event)
-	event.virus_type = virus_type
+	if(max_severity == ADV_DISEASE_HARMFUL)
+		possible_symptoms += list(
+			/datum/symptom/disfiguration,
+			/datum/symptom/hallucigen,
+			/datum/symptom/polyvitiligo,
+			/datum/symptom/visionloss,
+		)
 
-/datum/event_admin_setup/disease_outbreak/advanced
-	///Admin selected custom severity rating for the event
-	var/max_severity
-	///Admin selected custom value for the maximum symptoms this virus should have
-	var/max_symptoms
+	if(max_severity == ADV_DISEASE_DANGEROUS)
```

Comment:
switch(max_severity) might work better here so we don't have to evaluate the value twice.